lib/Runtime/Library/JSON.cpp

```diff
@@ -336,17 +332,28 @@ namespace JSON
 
     // -------- StringifySession implementation ------------//
 
+    static char16  gapStringBuffer[JSONspaceSize];
+    static char16* GetGapStringInternal()
+    {
+        wmemset(gapStringBuffer, _u(' '), JSONspaceSize);
+        return gapStringBuffer;
+    }
+
     void StringifySession::CompleteInit(Js::Var space, ArenaAllocator* tempAlloc)
     {
         //set the stack, gap
-        char16 buffer[JSONspaceSize];
-        wmemset(buffer, _u(' '), JSONspaceSize);
+        static const char16* spaceBuffer = GetGapStringInternal(); // multi threading safe
+
         charcount_t len = 0;
         switch (Js::JavascriptOperators::GetTypeId(space))
         {
         case Js::TypeIds_Integer:
             {
                 len = max(0, min(JSONspaceSize, static_cast<int>(Js::TaggedInt::ToInt32(space))));
+                if (len)
```

Comment:
my point is you are using int as bool expression in the if condition.

---
In reply to: [131467627](https://github.com/Microsoft/ChakraCore/pull/3455#discussion_r131467627) [](ancestors = 131467627)